code/datums/components/itempicky.dm

```diff
@@ -5,13 +5,17 @@
 	var/whitelist
 	/// Message shown if you try to pick up an item not in the whitelist
 	var/message = "You don't like %TARGET, why would you hold it?"
+	/// An optional callback we check for overriding our whitelist
+	var/datum/callback/tertiary_condition = null
 
-/datum/component/itempicky/Initialize(whitelist, message)
+/datum/component/itempicky/Initialize(whitelist, message, tertiary_condition)
 	if(!ismob(parent))
 		return COMPONENT_INCOMPATIBLE
 	src.whitelist = whitelist
 	if(message)
 		src.message = message
+	if(tertiary_condition)
+		src.tertiary_condition = tertiary_condition
```

Comment:
one more thing, null out tertiary_condition on Destory so we don't hang a ref to our parent/the contents of the callback